pkg/config/policy.go

```diff
@@ -0,0 +1,249 @@
+)
+
+// GetPoliciesForConsortiums returns a map of policies for channel consortiums.
+func GetPoliciesForConsortiums(config cb.Config) (map[string]*Policy, error) {
+	consortiums, ok := config.ChannelGroup.Groups[ConsortiumsGroupKey]
+	if !ok {
+		return nil, errors.New("consortiums missing from config")
+	}
+
+	return getPolicies(consortiums.Policies)
+}
+
+// GetPoliciesForConsortium returns a map of policies for a specific consortium.
+func GetPoliciesForConsortium(config cb.Config, consortiumName string) (map[string]*Policy, error) {
+	consortium, ok := config.ChannelGroup.Groups[ConsortiumsGroupKey].Groups[consortiumName]
+	if !ok {
+		return nil, fmt.Errorf("consortium %s does not exist in channel config", consortiumName)
+	}
+
+	return getPolicies(consortium.Policies)
+}
+
+// GetPoliciesForConsortiumOrg returns a map of policies for a specific consortium org.
+func GetPoliciesForConsortiumOrg(config cb.Config, consortiumName, orgName string) (map[string]*Policy, error) {
+	org, ok := config.ChannelGroup.Groups[ConsortiumsGroupKey].Groups[consortiumName].Groups[orgName]
+	if !ok {
+		return nil, fmt.Errorf("consortium org %s does not exist in channel config", orgName)
+	}
+
+	return getPolicies(org.Policies)
+}
+
+// GetPoliciesForOrderer returns a map of policies for channel orderer.
+func GetPoliciesForOrderer(config cb.Config) (map[string]*Policy, error) {
+	orderer, ok := config.ChannelGroup.Groups[OrdererGroupKey]
+	if !ok {
+		return nil, errors.New("orderer missing from config")
+	}
+
+	return getPolicies(orderer.Policies)
+}
+
+// GetPoliciesForOrdererOrg returns a map of policies for a specific consortium org.
+func GetPoliciesForOrdererOrg(config cb.Config, consortiumName, orgName string) (map[string]*Policy, error) {
+	org, ok := config.ChannelGroup.Groups[OrdererGroupKey].Groups[orgName]
+	if !ok {
+		return nil, fmt.Errorf("orderer org %s does not exist in channel config", orgName)
+	}
+
+	return getPolicies(org.Policies)
+}
+
+// GetPoliciesForApplication returns a map of policies for application config group.
+func GetPoliciesForApplication(config cb.Config) (map[string]*Policy, error) {
+	application, ok := config.ChannelGroup.Groups[ApplicationGroupKey]
+	if !ok {
+		return nil, errors.New("application missing from config")
+	}
+
+	return getPolicies(application.Policies)
+}
+
+// GetPoliciesForApplicationOrg returns a map of policies for specific application
+// organization.
+func GetPoliciesForApplicationOrg(config cb.Config, orgName string) (map[string]*Policy, error) {
+	orgGroup, ok := config.ChannelGroup.Groups[ApplicationGroupKey].Groups[orgName]
+	if !ok {
+		return nil, fmt.Errorf("application org %s does not exist in channel config", orgName)
+	}
+
+	return getPolicies(orgGroup.Policies)
+}
+
+// getPolicies returns a map of Policy from given map of ConfigPolicy in organization config group.
+func getPolicies(policies map[string]*cb.ConfigPolicy) (map[string]*Policy, error) {
+	if policies == nil {
```

Comment:
Why bother with this block? Ranging over nil results in 0 executions of the block. This just introduces an unnecessary path.

Reply:
the intent of this block is when policies are nil in given config group, we want to return a empty policy map instead of error out. 
but seems the original code already handles that. 
This block will be removed when fixes applied on top.